commands.js

```diff
@@ -2956,6 +2956,19 @@ exports.commands = {
 			Ladders(target).getTop().then(result => {
 				connection.send('|queryresponse|laddertop|' + JSON.stringify(result));
 			});
+		} else if (cmd === 'userlist') {
+			if (!trustable) return false;
+
+			let userlist = [];
+
+			for (let i in room.users) {
+				if (!room.users[i].named) {
+					continue;
+				}
+				userlist.push(room.users[i].userid);
+			}
+
+			connection.send(room + '|queryresponse|userlist|' + JSON.stringify(userlist));
```

Comment:
Well this sends the message to that room. I could make this work better
though and more in line with the other queries. Will do that actually.
Op 11 jun. 2016 05:13 schreef "panpawn" <notifications@github.com>:

> In commands.js
> <https://github.com/Zarel/Pokemon-Showdown/pull/2582#discussion_r66699872>
> :
>
> > @@ -2956,6 +2956,19 @@ exports.commands = {
> >  			Ladders(target).getTop().then(result => {
> >  				connection.send('|queryresponse|laddertop|' + JSON.stringify(result));
> >  			});
> > +		} else if (cmd === 'userlist') {
> > +			if (!trustable) return false;
> > +
> > +			let userlist = [];
> > +
> > +			for (let i in room.users) {
> > +				if (!room.users[i].named) {
> > +					continue;
> > +				}
> > +				userlist.push(room.users[i].userid);
> > +			}
> > +
> > +			connection.send(room + '|queryresponse|userlist|' + JSON.stringify(userlist));
>
> imo it should be |queryresponse|' + room + '|userlist| ...
>
> not that it matters much
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/Zarel/Pokemon-Showdown/pull/2582/files/13c1e08ec6f12220086dac85609e7fb86a9bb4c4#r66699872>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/AEQEcG7HOzHJBN7PqT4vvamTr4Fgxgq4ks5qKie7gaJpZM4Izc8i>
> .
>